litestar/response/streaming.py

```diff
@@ -3,12 +3,14 @@
 import itertools
 from collections.abc import AsyncGenerator, AsyncIterable, AsyncIterator, Iterable, Iterator
 from functools import partial
-from typing import TYPE_CHECKING, Any, Callable, Union
+from typing import TYPE_CHECKING, Any, Callable, TypeVar, Union
 
-from anyio import CancelScope, create_task_group
+from anyio import Event, create_task_group
 
 from litestar.enums import MediaType
+from litestar.exceptions import ClientDisconnectException
 from litestar.response.base import ASGIResponse, Response
+from litestar.response.sse import ServerSentEventIterator
```

Comment:
This module shouldn't need to know about `ServerSentEventIterator`. It should be compatible with all iterators / generators generally